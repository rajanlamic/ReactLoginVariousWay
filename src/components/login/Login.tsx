import React from 'react'

class Login extends React.Component<any, any> {
    constructor(props: any) {
        super(props)
        this.state = { userName: '', password: '' }
    }

    handleChange = (event: React.ChangeEvent<HTMLInputElement>, name: string) => {
        const value = event.target.value
        this.setState(() => {
            return {
                [name]: value
            }
        })
    }

    handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault()
        console.log('username', this.state.userName)
        console.log('password', this.state.password)
    }

    render() {
        return <form onSubmit={this.handleSubmit}>
            <div>Login with class</div>
            <label>Username</label>
            <input type="text" name="userName" value={this.state.userName} onChange={(event) => this.handleChange(event, "userName")} />
            <label>Password</label>
            <input type="password" name="password" value={this.state.password} onChange={(event) => this.handleChange(event, "password")} />
            <input type="submit" value="Submit" />
        </form>
    }

}

export default Login
