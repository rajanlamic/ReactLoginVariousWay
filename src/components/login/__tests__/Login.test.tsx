// import React from 'react'
import React from 'react'
import Login from '../Login'
import { shallow } from 'enzyme';

describe("Login", () => {
    let wrapper: any, props: any;
    beforeEach(() => {
        props = {

        }
        wrapper = shallow(<Login {...props} />)
    })

    afterEach(() => {
        props = null
        wrapper = null
    })

    it("should render properly matching with its snapshots", () => {
        expect(wrapper).toMatchSnapshot()
    })

    it("should render Login comnponner property", () => {
        expect(wrapper.find(".login-form").length).toEqual(1)
    })

    it("should render username and password label", () => {
        expect(wrapper.find('label').first().props().children).toBe('Username')
        expect(wrapper.find('label').at(1).props().children).toBe('Password')
    })

    it("should render first input as userName input with various attributes", () => {
        expect(wrapper.find('input').first().props().type).toBe('text')
        expect(wrapper.find('input').first().props().name).toBe('userName')
        expect(wrapper.find('input').first().props().value).toBeFalsy()
        expect(wrapper.find('input').first().props().onChange).toBeInstanceOf(Function)
    })

    it("userName input should change value by respoinding to onChnage event handler", () => {
        const handleChange = jest.spyOn(wrapper.instance(), 'handleChange')
        wrapper.find('input').first().simulate('change', { target: { value: '' } })
        wrapper.find('input').at(1).simulate('change', { target: { value: '' } })
        expect(handleChange).toHaveBeenCalled()
    })

    it("should update userName and password value", () => {
        wrapper.find('input').first().simulate('change', { target: { value: 'rajan' } })
        wrapper.find('input').at(1).simulate('change', { target: { value: 'pass' } })
        expect(wrapper.find('input').first().props().value).toBe('rajan')
        expect(wrapper.find('input').at(1).props().value).toBe('pass')
    })
})
