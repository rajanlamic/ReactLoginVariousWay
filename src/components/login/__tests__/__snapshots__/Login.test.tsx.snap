// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render properly matching with its snapshots 1`] = `ShallowWrapper {}`;
